File: RRP_Euler3/LargestPrimeFactor.py
import math as m

largest_prime = 1
target_value = 600851475143

def is_prime(num):
    for x in range(2, int(m.sqrt(num))):
        if num%x == 0:
            return False
    return True

print("here we go")
for x in range(2, target_value):
    if (target_value % x == 0) and is_prime(x):
        largest_prime = x
        print(largest_prime)

File: 3/LargestPrimeFactor.js
/*
I spent all afternoon debugging this in Node JS...
because I used var instead of let in my for loops...
sadge
I remade it in python because I didn't know why it didn't work
*/
var largestPrime = 1; // Largest prime factor
var targetVal = 600851475143; //600851475143

function isPrime(num) {
    for (let k = 2; k < parseInt(Math.sqrt(num) + 1); k++) {
        if (num%k === 0) {
            return false;
        }
    }
    return true;
}

console.log("here we go");
for (let i = 2; i < targetVal; i++) {
    if (targetVal % i === 0 && isPrime(i)){
        largestPrime = i;
        console.log(largestPrime);
    }
}